pypck/connection.py

```diff
@@ -456,8 +466,8 @@ def get_address_conn(
             else:
                 address_conn = ModuleConnection(self, addr)
                 if request_serials:
-                    self.request_serials_task = asyncio.create_task(
-                        address_conn.request_serials()
+                    self.request_serials_tasks.add(
+                        asyncio.create_task(address_conn.request_serials())
```

Comment:
My suggestion would be to replace this with:
```python
request_task = asyncio.create_task(address_conn.request_serials())
request_task.add_done_callback(
    lambda task: self.request_serials_tasks.remove(task)
)
self.request_serials_tasks.add(request_task)
```

And remove everything which has to do with the `cleanup_completed_tasks()` coroutine. This works for me also in the tests. Sure, we still have to cancel remaining tasks in `self.requests_serial_tasks` in `async_close()` using `gather`.

Reply:
Sure. I'll try it out and add a few more tests to check the done and not done cases.

Reply:
Works for me as well.  The new test should cover all cases.